Exercise_2.java

```diff
@@ -8,31 +8,69 @@ static class StackNode {
   
         StackNode(int data) 
         { 
-            //Constructor here 
+            this.data = data;
+            this.next = null;
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
         //Write your code here for the condition if stack is empty. 
+        if(root==null)
+            return true;
+        return false;
     } 
   
     public void push(int data) 

```

Comment:
You are implementing stack so push/pop node at head of the list so then out will be called stack.

Reply:
And you will be doing push/pop in O(1) time. IN your approach, it takes O(N) to do these operations.